Jenkinsfile

```diff
@@ -1,190 +1,139 @@
-            sh(returnStdout: true, script: "git config branch.main.remote origin")
-            sh(returnStdout: true, script: "git config branch.main.merge refs/heads/main")
+              // The git config setup required for your project prior to running 'sbt release with-defaults' may vary:
+              sh(returnStdout: true, script: "git config remote.origin.fetch +refs/heads/*:refs/remotes/origin/*")
+              sh(returnStdout: true, script: "git config branch.main.remote origin")
+              sh(returnStdout: true, script: "git config branch.main.merge refs/heads/main")
 
-            // increasing meta space size to avoid build errors during release step
-            javaopts = "JAVA_OPTS=-XX:MaxMetaspaceSize=512m"
+              // increasing meta space size to avoid build errors during release step
+              javaopts = "JAVA_OPTS=-XX:MaxMetaspaceSize=512m"
 
-            withEnv([javaopts]) {
-              echo sh(returnStdout: true, script: "echo y | sbt \"release with-defaults\"")
+              withEnv([javaopts]) {
+                echo sh(returnStdout: true, script: "echo y | sbt \"release with-defaults\"")
+              }
             }
-
-            cutNeeded = true
           }
+          echo 'Getting release tag'
+          release_tag = sh(returnStdout: true, script: "git describe --abbrev=0 --match \"v*\"").trim()
+          branchSpecifier = "refs/tags/${release_tag}"
+          echo branchSpecifier
 
-          if(cutNeeded == true) {
-            echo 'Getting release tag'
-            release_tag = sh(returnStdout: true, script: "git describe --abbrev=0 --match \"v*\"").trim()
-            branchSpecifier = "refs/tags/${release_tag}"
-            echo branchSpecifier
-
-            // checkout the tag so we're performing subsequent actions on it
-            sh "git checkout ${branchSpecifier}"
-
-            // set the service_sha to the current tag because it might not be the same as env.GIT_COMMIT
-            service_sha = sh(returnStdout: true, script: "git rev-parse HEAD").trim()
-
-            // set later stages to run since we're cutting
-            stage_build = true
-            stage_dockerize = true
-            stage_deploy = true
-          }
+          // checkout the tag so we're performing subsequent actions on it
+          sh "git checkout ${branchSpecifier}"
         }
       }
     }
     stage('Build') {
-      when { expression { stage_build } }
       steps {
         script {
           // perform any needed modifiers on the build parameters here
           sbtbuild.setRunITTest(true)
           sbtbuild.setNoSubproject(true)
-          sbtbuild.setScalaVersion(scala_version)
+          sbtbuild.setScalaVersion(env.SCALA_VERSION)
 
           // build
           echo "Building sbt project..."
           sbtbuild.build()
 
-          // If we're cutting, also publish libraries if they haven't already been published for this version
-          if(stage_cut == true) {
```

Comment:
I moved this to a Publish stage which is triggered only in release builds if there are files changed (other than version.sbt)